x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvSort.java

```diff
@@ -148,9 +150,21 @@ public EvalOperator.ExpressionEvaluator.Factory toEvaluator(
         Function<Expression, EvalOperator.ExpressionEvaluator.Factory> toEvaluator
     ) {
         Expression nonNullOrder = order == null ? ASC : order;
-        boolean ordering = nonNullOrder.foldable() && ((BytesRef) nonNullOrder.fold()).utf8ToString().equalsIgnoreCase("DESC")
-            ? false
-            : true;
+        boolean ordering = true;
+        if (nonNullOrder.foldable()) {
```

Comment:
I think we should throw if this isn't the case here, since `validate()` asks for it.

Reply:
Maybe this should actually be moved entirely there.

Reply:
Discussed with Bogdan, when the inputs to mv_sort are both constants/literals, logical plan optimizer's constant folding with convert the mv_sort to a constant, in this case validate() is not called, the check/throw here is for this situation.

```
[2024-06-24T04:15:35,634][TRACE][o.e.x.e.o.LogicalPlanOptimizer] [test] Rule rules.ConstantFolding applied
Limit[1000[INTEGER]]                                              = Limit[1000[INTEGER]]
\_Eval[[MVSORT([4, 2, 2][INTEGER],[44 45 53 43][KEYWORD]) AS sd]] ! \_Eval[[[4, 2, 2][INTEGER] AS sd]]
  \_Row[[[4, 2, -3, 2][INTEGER] AS a]]                            =   \_Row[[[4, 2, -3, 2][INTEGER] AS a]]
```